### DISCUSSION TITLE
How to remove/disable .watch() ?

### DISCUSSION THREAD

**Discussion by AlexShniperson-INO** [simolus3/drift#1654] [Q&A] [answered]
Hi
Is there a way to disable/remove/close created `.watch()`?
Looks like it's lack of documentation

**Comment by simolus3** [comment, marked as answer]
The internal query stream handler implements the [Stream](https://api.dart.dev/stable/2.15.1/dart-async/Stream-class.html) interface from the Dart SDK, and it behaves how one would expect: Each call to `.listen` returns a new `StreamSubscription`, and a watched stream will be disposed when all subscriptions have been cancelled. Streams also end when the database is closed.

I can give you more details if you're interested in something specific, but from a user's perspective there isn't much more to know about disposing watched streams, as they can be disposed just like any other well-behaving stream.

**Reply by AlexShniperson-INO** [reply]
so the way of solution for me is to make `.listen(...)` then collect `StreamSubscription`'s and then cancel them, is that correct?

**Reply by simolus3** [reply]
Yes exactly. In general, you should always make sure that you cancel `StreamSubscriptions` you create somehow, regardless of where the stream is coming from :) For widgets like a `StreamBuilder`, the framework will do this for you automatically. But in general, each call to `.listen` should have another call to `.cancel` somewhere.

Drift will dispose a query stream once all stream subscriptions have been cancelled.

**Reply by AlexShniperson-INO** [reply]
ok thank you for explanation
